rcli/typing.py

```diff
@@ -0,0 +1,673 @@
+                        raise ValueError(
+                            'The value of {}({}) [{}] is below the minimum '
+                            'allowed value of {}.'.format(
+                                keyfunc_name, repr(__value), repr(expected),
+                                bound.start))
+                    raise ValueError(
+                        'The value {} is below the minimum allowed value '
+                        'of {}.'.format(repr(__value), bound.start))
+                if bound.stop and expected > bound.stop:
+                    if keyfunc is not identity:
+                        raise ValueError(
+                            'The value of {}({}) [{}] is above the maximum'
+                            ' allowed value of {}.'.format(
+                                keyfunc_name, repr(__value), repr(expected),
+                                bound.stop))
+                    raise ValueError(
+                        'The value {} is above the maximum allowed value '
+                        'of {}.'.format(repr(__value), bound.stop))
+                return ret
+
+        if keyfunc is not cls._default:
+            _BoundedSubclass.__class_repr__ = '{}.{}[{}, {}, {}]'.format(
+                cls.__module__, cls.__name__, cls._get_fullname(type_),
+                cls._get_bound_repr(bound), keyfunc_name)
+        else:
+            _BoundedSubclass.__class_repr__ = '{}.{}[{}, {}]'.format(
+                cls.__module__, cls.__name__, cls._get_fullname(type_),
+                cls._get_bound_repr(bound))
+        return _BoundedSubclass
+
+    def _get_args(cls, args):
+        # type: (tuple) -> Tuple[Any, slice, Callable]
+        """Return the parameters necessary to check type boundaries.
+
+        Args:
+            args: A tuple with two or three parameters: a type, a slice
+                representing the minimum and maximum lengths allowed for values
+                of that type and, optionally, a function to use on values
+                before comparing against the bounds.
+
+        Returns:
+            A tuple with three parameters: a type, a slice, and a function to
+            apply to objects of the given type. If no function was specified,
+            it returns the identity function.
+        """
+        if not isinstance(args, tuple):
+            raise TypeError(
+                '{}[...] takes at least two arguments.'.format(cls.__name__))
+        elif len(args) == 2:
+            type_, bound = args
+            keyfunc = cls._identity
+        elif len(args) == 3:
+            type_, bound, keyfunc = args
+        else:
+            raise TypeError(
+                'Too many parameters given to {}[...]'.format(cls.__name__))
+        if not isinstance(bound, slice):
+            bound = slice(bound)
+        if isinstance(type_, six.string_types):
+            # pragma pylint: disable=protected-access
+            ref = _ForwardRef(type_)
```

Comment:
I feel like these two lines would actually be slightly more readable if they were combined.